src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -1076,8 +1099,9 @@ fn generic_simd_intrinsic(
     }
 
     let tcx = bx.tcx();
+    let param_env = ty::ParamEnv::reveal_all();
     let sig = tcx.normalize_erasing_late_bound_regions(
-        ty::ParamEnv::reveal_all(),
+        param_env,
```

Comment:
Is this change even needed anymore?

Reply:
No, I don't think so. This was some downfall of the previous code that was calling the now removed `simd_size` and was passing `param_env` to it everywhere.